common/achievement/create.ts

```diff
@@ -1,29 +1,37 @@
-import { Achievement_template } from '../../graphql/generated';
+import { Achievement_template, JsonFilter } from '../../graphql/generated';
+import { ActionID, SpecificNotificationContext } from '../notification/actions';
 import { prisma } from '../prisma';
 import { TemplateSelectEnum, getAchievementTemplates } from './template';
-import { UserAchievementContext } from './types';
 
-async function doesUserAchievementAlreadyExist(templateId: number, userId: string) {
-    // TODO - check if user achievement exist for one match or one subcourse
+async function doesUserAchievementAlreadyExist<ID extends ActionID>(templateId: number, userId: string, context: SpecificNotificationContext<ID>) {
+    const keys = Object.keys(context);
     const userAchievement = await prisma.user_achievement.findFirst({
         where: {
             templateId,
             userId,
+            AND: keys.map((key) => {
```

Comment:
So with this check we can guarantee that the user_achievement for a user exists for a specific match / subcourse?